File: ant-sdk/antd-php/examples/04-files.php
<?php

/**
 * Example 04: Upload and download files publicly, with round-trip assertions.
 *
 * Prerequisites: antd daemon running with a funded wallet.
 */

require_once __DIR__ . '/../vendor/autoload.php';

use Autonomi\Antd\AntdClient;

function rrmdir(string $dir): void {
    if (!is_dir($dir)) {
        if (file_exists($dir)) {
            unlink($dir);
        }
        return;
    }
    foreach (scandir($dir) as $entry) {
        if ($entry === '.' || $entry === '..') continue;
        $path = $dir . DIRECTORY_SEPARATOR . $entry;
        is_dir($path) ? rrmdir($path) : unlink($path);
    }
    rmdir($dir);
}

$tmp = sys_get_temp_dir() . DIRECTORY_SEPARATOR . 'antd-php-04-files';
rrmdir($tmp);
mkdir($tmp, 0o700, true);

$fileContent = "Hello from a file on Autonomi!";

$srcFile = $tmp . '/hello.txt';
file_put_contents($srcFile, $fileContent);

$client = new AntdClient();

$cost = $client->fileCost($srcFile, true);
echo "Estimated cost: {$cost->cost} atto ({$cost->chunkCount} chunks)\n";

$result = $client->filePutPublic($srcFile);
echo "File uploaded at: {$result->address}\n";
echo "  storage: {$result->storageCostAtto} atto, gas: {$result->gasCostWei} wei\n";
echo "  chunks: {$result->chunksStored}, mode: {$result->paymentModeUsed}\n";

$dstFile = $tmp . '/hello.txt.downloaded';
$client->fileGetPublic($result->address, $dstFile);
echo "File downloaded to {$dstFile}\n";

if (file_get_contents($dstFile) !== $fileContent) {
    rrmdir($tmp);
    fwrite(STDERR, "round-trip mismatch on hello.txt\n");
    exit(1);
}

rrmdir($tmp);
echo "File upload/download OK!\n";

File: ant-sdk/antd-dart/example/04_files.dart
import 'dart:io';
import 'package:antd/antd.dart';

/// Demonstrates file upload/download with a real tempfile + round-trip assertion.
Future<void> main() async {
  final client = AntdClient();
  final tmp = await Directory.systemTemp.createTemp('antd-dart-04-files-');

  try {
    const fileContent = 'Hello from a file on Autonomi!';
    final srcFile = File('${tmp.path}/hello.txt');
    await srcFile.writeAsString(fileContent);

    final cost = await client.fileCost(srcFile.path);
    print('Estimated cost: ${cost.cost} atto across ${cost.chunkCount} chunks');

    final result = await client.filePutPublic(srcFile.path);
    print('File uploaded at ${result.address}');
    print('  storage: ${result.storageCostAtto} atto, gas: ${result.gasCostWei} wei');
    print('  chunks: ${result.chunksStored}, mode: ${result.paymentModeUsed}');

    final dstFile = File('${tmp.path}/hello.txt.downloaded');
    await client.fileGetPublic(result.address, dstFile.path);
    print('File downloaded to ${dstFile.path}');

    final got = await dstFile.readAsString();
    if (got != fileContent) {
      stderr.writeln('round-trip mismatch on hello.txt');
      exitCode = 1;
      return;
    }

    print('File upload/download OK!');
  } finally {
    client.close();
    await tmp.delete(recursive: true);
  }
}
